combat/CombatEvents.cpp

```diff
@@ -763,73 +763,104 @@ void IncapacitationEvent::serialize<freeorion_xml_oarchive>(freeorion_xml_oarchi
 template
 void IncapacitationEvent::serialize<freeorion_xml_iarchive>(freeorion_xml_iarchive& ar, const unsigned int version);
 
+
 //////////////////////////////////////////
-///////// FighterAttackedEvent ////////
+///////// FightersAttackFightersEvent ////
 //////////////////////////////////////////
-FighterAttackedEvent::FighterAttackedEvent() :
+
+FightersAttackFightersEvent::FightersAttackFightersEvent() :
     bout(-1),
-    round(-1),
-    attacker_owner_empire_id(ALL_EMPIRES),
-    attacked_by_object_id(INVALID_OBJECT_ID),
-    attacked_owner_id(ALL_EMPIRES)
+    events()
 {}
 
-FighterAttackedEvent::FighterAttackedEvent(int bout_, int round_, int attacked_by_object_id_,
-                                                 int attacker_owner_empire_id_, int attacked_owner_id_) :
+FightersAttackFightersEvent::FightersAttackFightersEvent(int bout_) :
     bout(bout_),
-    round(round_),
-    attacker_owner_empire_id(attacker_owner_empire_id_),
-    attacked_by_object_id(attacked_by_object_id_),
-    attacked_owner_id(attacked_owner_id_)
+    events()
 {}
 
-std::string FighterAttackedEvent::DebugString() const {
+void FightersAttackFightersEvent::AddEvent(int attacker_empire_, int target_empire_)
+{ events[{attacker_empire_, target_empire_}] += 1; }
+
+std::string FightersAttackFightersEvent::DebugString() const {
     std::stringstream ss;
-    ss << "rnd: " << round << " : "
-       << attacked_by_object_id << " -> (Fighter of Empire " << attacked_owner_id << ")"
-       << " by object owned by empire " << attacker_owner_empire_id
-       << " at bout " << bout << " round " << round;
+    ss << "FightersAttackFightersEvent: ";
+    for (const auto& index_and_event: events) {
+        ss << index_and_event.second << " repeated fighters from empire " << index_and_event.first.first
+           << " attacking fighters from empire " << index_and_event.first.second << ", ";
+    }
     return ss.str();
 }
 
-std::string FighterAttackedEvent::CombatLogDescription(int viewing_empire_id) const {
-    std::string attacked_by = FighterOrPublicNameLink(viewing_empire_id, attacked_by_object_id, attacker_owner_empire_id);
-    std::string empire_coloured_attacked_fighter = EmpireColorWrappedText(attacked_owner_id, UserString("OBJ_FIGHTER"));
+std::string FightersAttackFightersEvent::CombatLogDescription(int viewing_empire_id) const {
+    if (events.empty())
+        return "";
+
+    const auto& events_to_show = events;
+    auto num_events_remaining = events.size();
+    std::stringstream ss;
 
-    const std::string& template_str = UserString("ENC_COMBAT_ATTACK_SIMPLE_STR");
+    // Use show_events_for_empire to show events in this order: viewing empire, ALL_EMPIRES and
+    // then the remainder.
+    auto show_events_for_empire = [&ss, &num_events_remaining, &events_to_show, &viewing_empire_id](boost::optional<int> show_attacker)
```

Comment:
there's not really a specific precedent for this type of line, but this one seems too long and should probably be broken over several